invenio_app_ils/ill/api.py

```diff
@@ -88,11 +91,57 @@ def delete(self, **kwargs):
 )
 
 
+class IllValidator(RecordValidator):
+    """Ill record validator."""
+
+    def validate_cancel(self, status, cancel_reason):
+        """Validate decline is correct."""
+        if status == "CANCELLED" and not cancel_reason:
+            raise ILLError(
+                "You have to provide a cancel reason when cancelling a request"
+            )
+        if cancel_reason and not status == "CANCELLED":
+            raise ILLError(
+                "If you select a cancel reason you need to select"
+                " \"Cancelled\" in the state"
+            )
+
+    def ensure_document_exists(self, document_pid):
+        """Ensure document exists or raise."""
+        Document = current_app_ils.document_record_cls
+
+        try:
+            Document.get_record_by_pid(document_pid)
+        except PIDDoesNotExistError:
+            raise DocumentNotFoundError(document_pid)
+
+    def ensure_library_exists(self, library_pid):
+        """Ensure library exists or raise."""
+        try:
+            Library.get_record_by_pid(library_pid)
+        except PIDDoesNotExistError:
+            raise LibraryNotFoundError(library_pid)
+
+    def validate(self, record, **kwargs):
+        """Validate record before create and commit."""
+        super().validate(record, **kwargs)
+
+        status = record.get("status", None)
```

Comment:
Mandatory field:
```suggestion
        status = record["status"]
```
can you please do the same from document and library?